File: src/main/java/io/github/jsnimda/inventoryprofiles/compat/ModMenuApiImpl.java
package io.github.jsnimda.inventoryprofiles.compat;

import io.github.jsnimda.inventoryprofiles.gui.ConfigScreen;
import io.github.prospector.modmenu.api.ConfigScreenFactory;
import io.github.prospector.modmenu.api.ModMenuApi;

/**
 * ModMenuApiImpl
 */
public class ModMenuApiImpl implements ModMenuApi {

  @Override
  public ConfigScreenFactory<?> getModConfigScreenFactory() {
    return ConfigScreen::new;
  }

}

File: src/main/java/io/github/jsnimda/inventoryprofiles/compat/ModMenuApiImpl.kt
package io.github.jsnimda.inventoryprofiles.compat

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import io.github.jsnimda.common.vanilla.alias.Screen
import io.github.jsnimda.inventoryprofiles.gui.ConfigScreen
import com.terraformersmc.modmenu.api.ModMenuApi
import java.util.function.Function

class ModMenuApiImpl : ModMenuApi {

  override fun getModConfigScreenFactory(): ConfigScreenFactory<ConfigScreen> {
    return ConfigScreenFactory<ConfigScreen> { parent ->
      val c = ConfigScreen()
      c.parent = parent
      c
    }
  }

    /*

     { parent: Screen ->
      ConfigScreen().apply {
        this.parent = parent
      }
    }

     */
}